src/TLuaInterpreter.cpp

```diff
@@ -1787,10 +1789,27 @@ int TLuaInterpreter::saveMap(lua_State* L)
         }
     }
 
+    if (lua_gettop(L) == 2) {
+        if (!lua_isstring(L, 1)) {
+            lua_pushstring(L, "saveMap: wrong argument location.");
+            lua_error(L);
+            return 1;
+        } else {
+            location = lua_tostring(L, 1);
+        }
+        if (!lua_isnumber(L, 2)) {
+            lua_pushstring(L, "saveMap: wrong argument version.");
```

Comment:
New pattern error message would be:

    lua_pushfstring(L, "saveMap: bad argument #2 type (map format version as optional integer expected, got %s!)", luaL_typename(L, 2))l